apps/test/unit/templates/sectionProgress/progressTables/ProgressTableDetailCellTest.jsx

```diff
@@ -0,0 +1,177 @@
+import {expect} from '../../../../util/reconfiguredChai';
+import {shallow} from 'enzyme';
+import ProgressTableDetailCell, {
+  unitTestExports
+} from '@cdo/apps/templates/sectionProgress/progressTables/ProgressTableDetailCell';
+import ProgressTableLevelBubble from '@cdo/apps/templates/sectionProgress/progressTables/ProgressTableLevelBubble';
+import sinon from 'sinon';
+import firehoseClient from '@cdo/apps/lib/util/firehose';
+import * as progressStyles from '@cdo/apps/templates/progress/progressStyles';
+import i18n from '@cdo/locale';
+
+const LEVEL_1 = {
+  isUnplugged: true,
+  sublevels: undefined,
+  url: '/level-1',
+  id: '123',
+  bonus: null,
+  isConceptLevel: true,
+  bubbleText: '1',
+  kind: 'puzzle',
+  levelNumber: 1
+};
+
+const SUBLEVEL_1 = {
+  isUnplugged: true,
+  sublevels: undefined,
+  url: '/sublevel-1',
+  id: '789',
+  bonus: null,
+  isConceptLevel: true,
+  bubbleText: '1',
+  kind: 'puzzle',
+  levelNumber: 1
+};
+
+const LEVEL_2 = {
+  isUnplugged: false,
+  sublevels: [SUBLEVEL_1],
+  url: '/level-2',
+  id: '456',
+  bonus: null,
+  isConceptLevel: true,
+  bubbleText: '2',
+  kind: 'assessment',
+  levelNumber: 2
+};
+
+const LEVEL_3 = {
+  isUnplugged: false,
+  sublevels: undefined,
+  url: '/level-3',
+  id: '999',
+  bonus: true,
+  isConceptLevel: true,
+  bubbleText: '3',
+  kind: 'puzzel',
+  levelNumber: 3
+};
+
+const DEFAULT_PROPS = {
+  studentId: 1,
+  sectionId: 123,
+  levels: [LEVEL_1, LEVEL_2, LEVEL_3],
+  studentProgress: {
+    123: {pages: null, paired: false, result: 100, status: 'perfect'},
+    456: {pages: null, paired: false, result: 1001, status: 'locked'},
+    789: {pages: null, paired: false, result: 100, status: 'locked'}
+  },
+  stageExtrasEnabled: false
+};
+
+describe('ProgressTableDetailCell', () => {
+  beforeEach(() => {
+    sinon.stub(firehoseClient, 'putRecord');
+  });
+
+  afterEach(() => {
+    firehoseClient.putRecord.restore();
+  });
+
+  it('displays a progress table level bubble for each level and sublevel', () => {
+    const wrapper = shallow(<ProgressTableDetailCell {...DEFAULT_PROPS} />);
+    const levelBubble1 = wrapper.findWhere(node => node.key() === '123_1');
+    const levelBubble2 = wrapper.findWhere(node => node.key() === '456_2');
+    const levelBubble3 = wrapper.findWhere(node => node.key() === '999_3');
+    expect(levelBubble1.find(ProgressTableLevelBubble)).to.have.length(1);
+    expect(levelBubble2.find(ProgressTableLevelBubble)).to.have.length(2); // 1 for level, 1 for sublevel
+    expect(levelBubble3.find(ProgressTableLevelBubble)).to.have.length(1);
+  });
+
+  it('disables progress bubble if there is a bonus and stageExtrasEnabled is false', () => {
+    const wrapper = shallow(<ProgressTableDetailCell {...DEFAULT_PROPS} />);
+    const levelBubble3 = wrapper.findWhere(node => node.key() === '999_3');
+    const progressBubble = levelBubble3.find(ProgressTableLevelBubble);
+    expect(progressBubble.props().disabled).to.equal(true);
```

Comment:
now that i've fixed these in 1604, let's not merge any more instances in. `to.be.true` and `to.be.false` please!